src/System.Memory/src/System/MemoryExtensions.cs

```diff
@@ -10,13 +10,317 @@
 using Internal.Runtime.CompilerServices;
 #endif
 
+#if !netstandard11
+using System.Numerics;
+#endif
+
 namespace System
 {
     /// <summary>
     /// Extension methods for Span{T}, Memory{T}, and friends.
     /// </summary>
     public static partial class MemoryExtensions
     {
+        /// <summary>
+        /// Removes all leading and trailing white-space characters from the span.
+        /// </summary>
+        /// <param name="span">The span</param>
+        public static ReadOnlySpan<char> Trim(this ReadOnlySpan<char> span)
+        {
+            return span.Length == 0 ?
+                    default :
+                    span.TrimStart().TrimEnd();
+        }
+
+        /// <summary>
+        /// Removes all leading white-space characters from the span.
+        /// </summary>
+        /// <param name="span">The span</param>
+        public static ReadOnlySpan<char> TrimStart(this ReadOnlySpan<char> span)
+        {
+            int start = 0;
+            for (; start < span.Length; start++)
+            {
+                if (!char.IsWhiteSpace(span[start]))
+                    break;
+            }
+            return span.Slice(start);
```

Comment:
do we have to Slice if we didn't find any white spaces? same question for TrimEnd too.

Reply:
No, but checking start == 0 adds cost, so it is a trade off whether any character was trimmed or not. Either way, it is fairly negligible either way (< 5%).

Reply:
Shouldn't `span.Slice(0)` just be a noop anyway?